lib/stb_image.ex

```diff
@@ -103,6 +103,105 @@ defmodule StbImage do
         "unsupported tensor shape: #{inspect(shape)} (expected height-width-channel)"
       )
 
+  if Code.ensure_loaded?(Kino.Render) do
+    defimpl Kino.Render do
+      defp within_maximum_size(image) do
+        max_size = Application.fetch_env!(:stb_image, :kino_render_max_size)
+
+        case max_size do
+          {max_height, max_width} when is_integer(max_height) and is_integer(max_width) ->
+            case image.shape do
+              {h, w} ->
+                h <= max_height and w <= max_width
+
+              {h, w, _c} ->
+                h <= max_height and w <= max_width
+
+              _ ->
+                false
+            end
+
+          _ ->
+            raise """
+            invalid :kino_render_max_size configuration. Expected a 2-tuple, {height, width},
+            where height and width are both integers. Got: #{inspect(max_size)}
+            """
+        end
+      end
+
+      @spec to_livebook(StbImage.t()) :: Kino.Output.t()
+      def to_livebook(image) when is_struct(image, StbImage) do
+        render_types = Application.fetch_env!(:stb_image, :kino_render_tab_order)
+
+        Enum.map(render_types, fn
+          :raw ->
+            {"Raw", Kino.Inspect.new(image)}
+
+          :numerical ->
+            if Code.ensure_loaded?(Nx) do
+              {"Numerical", Kino.Inspect.new(StbImage.to_nx(image))}
+            else
+              {"Numerical",
+               Kino.Markdown.new("""
+               The `Numerical` tab requires application `:nx`, please add `{:nx, "~> 0.4"}` to the dependency list.
+               """)}
+            end
+
+          :image ->
+            render_encoding = Application.fetch_env!(:stb_image, :kino_render_encoding)
+
+            if render_encoding not in [:png, :jpg, :jpeg] do
+              raise "invalid :kino_render_encoding value. Expected one of :png, :jpg, or :jpeg. Got: #{inspect(render_encoding)}"
```

Comment:
We can move this to a default clause in the case below, where we already check all valid values :)